main.cpp

```diff
@@ -207,35 +197,68 @@ DOUBLE
 */
 struct DoubleType
 {
-    double add(double lhs, double rhs);
-    double subtract(double lhs, double rhs);
-    double multiply(double lhs, double rhs);
-    double divide(double lhs, double rhs);
+    DoubleType(double doubleValue)
+    {
+        value = new double(doubleValue);
+    }
```

Comment:
If you have to use a constructor that takes parameters, and those parameters initialize member variables, you should always prefer the member initialize list compared to initializing the variables inside the constructor body.
The reason is because the variables need to be instantiated/initialized when the class is created, and then when you assign to them in the ctor body, you are potentially re-initializing them.  that is a waste of resources.
- [ ] revise, use the member initializer list.